### PR TITLE
Feature/yarn assets

### PR DESCRIPTION
**Pull request by citizen428**
@hiattp As discussed previously, it's time to leave asset gems behind and go all in on Yarn. I removed both the jquery and bootstrap gems, and added them back via Yarn (including tether). Verified manually that everything is working as expected, including the bootstrap variable overrides.

@karunsiri Just added you as a reviewer because I figured you might be interested in this. 😺

### CODE REVIEW

**Comment by janklimo** [comment]
@citizen428 this is great 👏  Yarn ftw 🚀

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -11,8 +11,8 @@
 // about supported directives.
 //
 //= require rails-ujs
-//= require jquery3
 //= require turbolinks
-//= require popper
-//= require bootstrap-sprockets
+//= require jquery/dist/jquery
+//= require popper.js/dist/popper
```
Comment (roonglit) [inline app/assets/javascripts/application.js]: probably need to use umd version
`//= require popper.js/dist/umd/popper`

Comment (citizen428) [inline app/assets/javascripts/application.js]: Good catch @roonglit, 🙏

Comment (citizen428) [inline app/assets/javascripts/application.js]: Just for reference, the npm installed Bootstrap actually still uses Tether, so I switched out the dependency.

**Comment by karunsiri** [comment]
@citizen428 @roonglit I don't see `popper.js` in the changes, so I added it back and loaded the _UMD_ file in `application.js`

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -13,6 +13,7 @@
 //= require rails-ujs
 //= require turbolinks
 //= require jquery/dist/jquery
+//= require popper.js/dist/umd/popper
```
Comment (roonglit) [inline app/assets/javascripts/application.js]: @karunsiri I think Michael removed this off, and back to tether.

Comment (karunsiri) [inline app/assets/javascripts/application.js]: Ahh I see. I saw it listed in README, so I assume we're using it.

Comment (karunsiri) [inline app/assets/javascripts/application.js]: Please remove it if we don't need it. Thank you 😉

**Comment by citizen428** [comment]
@karunsiri I had removed it cause the version of Bootstrap in the npm package requires Tether and having both seemed like overkill.

**Comment by citizen428** [comment]
@karunsiri I squashed your commit into mine. I think the easiest thing to do for now is keeping Tether. People who prefer popper.js can do what @roonglit suggested above. Everyone else can just wait for Bootstrap 4 to become stable, they made popper the default in May.

**Pull request merged**